components/apimgt/org.wso2.carbon.apimgt.event.output.adapter.http.extended.oauth/src/main/java/org/wso2/carbon/apimgt/event/output/adapter/http/extended/oauth/ExtendedHTTPEventAdapter.java

```diff
@@ -0,0 +1,367 @@
+            } else if(oauthUrl != null && consumerKey != null && consumerSecret != null ) {
+                String accessToken = ServiceReferenceHolder.getInstance().getAccessTokenGenerator().getAccessToken(oauthUrl,consumerKey,
+                        consumerSecret);
+                executorService.submit(new HTTPSender(url, payload, accessToken, headers, httpClient));
+            }
+        } catch (RejectedExecutionException e) {
+            EventAdapterUtil
+                    .logAndDrop(eventAdapterConfiguration.getName(), message, "Job queue is full", e, log, tenantId);
+        }
+    }
+
+    @Override
+    public void disconnect() {
+        //not required
+    }
+
+    @Override
+    public void destroy() {
+        //not required
+    }
+
+    @Override
+    public boolean isPolled() {
+        return false;
+    }
+
+    private void checkHTTPClientInit(Map<String, String> staticProperties) {
+
+        if (this.httpClient != null) {
+            return;
+        }
+
+        synchronized (ExtendedHTTPEventAdapter.class) {
+            if (this.httpClient != null) {
+                return;
+            }
+
+            httpClient = new HttpClient(connectionManager);
+            String proxyHost = staticProperties.get(ExtendedHTTPEventAdapterConstants.ADAPTER_PROXY_HOST);
+            String proxyPort = staticProperties.get(ExtendedHTTPEventAdapterConstants.ADAPTER_PROXY_PORT);
+            if (proxyHost != null && proxyHost.trim().length() > 0) {
+                try {
+                    HttpHost host = new HttpHost(proxyHost, Integer.parseInt(proxyPort));
+                    this.httpClient.getParams().setParameter(ConnRoutePNames.DEFAULT_PROXY, host);
+                } catch (NumberFormatException e) {
+                    log.error("Invalid proxy port: " + proxyPort + ", "
+                            + "ignoring proxy settings for HTTP output event adaptor...");
+                }
+            }
+
+            String messageFormat = eventAdapterConfiguration.getMessageFormat();
+            if (messageFormat.equalsIgnoreCase("json")) {
+                contentType = "application/json";
+            } else if (messageFormat.equalsIgnoreCase("text")) {
+                contentType = "text/plain";
+            } else {
+                contentType = "text/xml";
+            }
+
+        }
+
+    }
+
+    private Map<String, String> extractHeaders(String headers) {
+        if (headers == null || headers.trim().length() == 0) {
+            return null;
+        }
+
+        String[] entries = headers.split(ExtendedHTTPEventAdapterConstants.HEADER_SEPARATOR);
+        String[] keyValue;
+        Map<String, String> result = new HashMap<String, String>();
+        for (String header : entries) {
+            try {
+                keyValue = header.split(ExtendedHTTPEventAdapterConstants.ENTRY_SEPARATOR, 2);
+                result.put(keyValue[0].trim(), keyValue[1].trim());
+            } catch (Exception e) {
```

Comment:
please explain reason to catch the generic exception.

Reply:
Removed the exception